### PR TITLE
feat: migrate fields to ACF Pro

### LINKED ISSUE #307: Switch to Advanced Custom Fields Pro
## Is your feature request related to a problem? Please describe.

ACF Pro provides built-in validation and will be easier to maintain.

## Describe the solution you'd like

Migrate existing custom fields from CMB2 to ACF Pro.

## Describe alternatives you've considered

Not applicable.

## Additional context

Not applicable.

### PR DESCRIPTION
**Pull request by greatislander**
* [x] I've read the [guidelines for Contributing to the Platform Co-op Toolkit](https://github.com/platform-coop-toolkit/.github/blob/master/CONTRIBUTING.md)
* [x] This isn't a duplicate of an existing pull request

## Description

Migrates custom fields to ACF Pro.

- [x] Create new fields.
- [x] Customize the `day` field based on selected `year` and `month`.
- [x] Save publication date if applicable.
- [x] Add validation for Perma.cc and Internet Archive links.
- [x] Add validation for DOI.
- [x] Add validation for ISBN.
- [x] Add validation for ISSN.
- [x] Convert existing authors, editors, and translators.
- [x] Convert existing Perma.cc and Internet Archive links.

## Steps to test

1. Check out this branch.
2. Update Composer dependencies: `composer install`.
3. Update NPM dependencies and build assets: `npm ci && npm run build`.
4. Install and activate Advanced Custom Fields Pro.
5. Run the [metadata conversion script](https://github.com/platform-coop-toolkit/coop-library-framework/blob/a02117529f10e87976152bf2eebbcb48323a3bab/bin/README.md).
5. Visit resource editor, test editing UI and user experience while modifying all fields.

**Expected behavior:** Validation is handled according to expectations and appropriate messages are displayed to users. Authors appear as expected in new author fields.

## Additional information

Not applicable.

## Related issues

- Resolves #307.

### CODE REVIEW

File: includes/functions/metadata.php [hunk 1]
```diff
@@ -363,483 +375,1055 @@ function register_meta() {
...
+
+	$host = wp_parse_url( $value, PHP_URL_HOST );
+
+	if ( 'web.archive.org' !== $host ) {
+		$valid = __( 'This is not a valid Internet Archive link.', 'coop-library-framework' );
 	}
 
-	return $options;
+	return $valid;
 }
```
Comment (jhung) [inline includes/functions/metadata.php]: Archive.org links may not have the `web.` subdomain prefix and causing this validator to fail.

Example:
```
https://archive.org/details/TheMacin1985
https://archive.org/details/in.gazette.central.e.2020-02-05.216023
```

**Review by jhung — CHANGES_REQUESTED** [review]
@jhung TODO:
- [ ] Check migration of Catalog IDs.
- [ ] Check migration of any archive.org and Perma.cc records

@greatislander see comment about archive.org validation

There are some other validation issues, but I will file those as separate issues.

File: includes/functions/metadata.php (hunk 1 above)
Comment (greatislander) [reply includes/functions/metadata.php]: @jhung resolved this in 21d1c1d.

Comment (greatislander) [reply includes/functions/metadata.php]: https://github.com/platform-coop-toolkit/coop-library-framework/pull/308/commits/21d1c1d7816f166c8873b535377e1af17a3dcb09#diff-c538c39054fe7d208782bb14bc3ee3d8R1424

**Review by jhung — APPROVED** [review]

**Pull request merged**